src/RefractDataStructure.cc

```diff
@@ -723,12 +760,22 @@ namespace drafter {
 
             if (property.node.name.variable.values.size() > 1) {
                 // FIXME: is there example for multiple variables?
-                throw snowcrash::Error("multiple variables in property definition are not allowed", snowcrash::MSONError);
+                throw snowcrash::Error("multiple variables in property definition are not implemented", snowcrash::MSONError);
             }
 
             snowcrash::SourceMap<mson::Literal> sourceMap;
             sourceMap.sourceMap.append(property.sourceMap.name.sourceMap);
 
+            // check if base variable type is StringElement
+            if (!property.node.name.variable.typeDefinition.empty()) {
+                const std::string& type = property.node.name.variable.typeDefinition.typeSpecification.name.symbol.literal;
+                if (!type.empty()) {
+                    if (!refract::TypeQueryVisitor::as<refract::StringElement>(FindRootAncestor(type, GetNamedTypesRegistry()))) {
```

Comment:
This is different case, *VariablePropertyName*
```
## VariableProperty (string)
### Sample
this can variate

## ObjectWithVariableProperty
- *var (VariableProperty)* (number)
```

and in MSON definition is written:
> In the case of specifying a Variable Property Name, the Value Definition MAY reference a Named Type that MUST be sub-typed from a string Primitive Type.